Lux/src/effect.cpp

```diff
@@ -0,0 +1,56 @@
+#include "effect.hpp"
+
+// In this case t has no effect
+template< class T > bool eff_vector_warp<T>::operator () ( buffer_pair< T > &buf, const float &t )  { 
+    if( buf.has_image() ) {
+        // Use buffer_pair operator () to return reference to first member of pair
+        buf.get_buffer().warp( buf(), vf, step, smooth, relative, extend );
+        buf.swap();
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+// Fill image with color
+template< class T > bool eff_fill<T>::operator () ( buffer_pair< T > &buf, const float &t )  { 
+    if( buf.has_image() ) {
+        if( bounded ) buf.get_image().fill( fill_color, bounds );
+        else          buf.get_image().fill( fill_color );
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+// Noise effect
+template< class T > bool eff_noise<T>::operator () ( buffer_pair< T > &buf, const float &t )  { 
+    if( buf.has_image() ) {
+        if( bounded ) buf.get_image().noise( a, bounds );
+        else          buf.get_image().noise( a );
+        return true;
```

Comment:
wonder if both returns should be merged into one at the bottom of the function. seems like this would eliminate most else clauses. just use ```return buf.has_image();``` once at end

Reply:
If there's no image in the buffer, I can't get the image or run and effect on it. It think it'll be more clear just to throw an exception.